src/NuGet.Core/NuGet.PackageManagement/IInstallationCompatibility.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Packaging.Core;
+using NuGet.ProjectManagement;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.PackageManagement
+{
+    /// <summary>
+    /// Validates the compatibility of a installed packages for the given project type. This
+    /// component should be used after packages have been downloaded to disk but have not yet
+    /// been installed to the project. If an installed package is not compatibile with the given
+    /// project, an exception is thrown.
+    /// </summary>
+    public interface IInstallationCompatibility
```

Comment:
This name should be specific to package type. The main compatibility check looks at the package assets, and this doesn't cover that.